pkg/sidecar/k8s_instance.go

```diff
@@ -454,3 +458,34 @@ func getRedisRoute(handle *netlink.Handle, redisIP net.IP) (*netlink.Route, erro
 
 	return &redisRoute, nil
 }
+
+func waitForPodRunningPhase(ctx context.Context, podName string) error {
+	k8scfg, err := clientcmd.BuildConfigFromFlags("", "")
+	if err != nil {
+		return fmt.Errorf("error in wait for pod running phase: %v", err)
+	}
+
+	k8sClientset, err := kubernetes.NewForConfig(k8scfg)
+	if err != nil {
+		return fmt.Errorf("error in wait for pod running phase: %v", err)
+	}
+
+	var phase string
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("wait for pod context (pod name: %s) erred with: %w", podName, ctx.Err())
+		default:
+			if phase == "Running" {
+				return nil
+			}
+			pod, err := k8sClientset.CoreV1().Pods("default").Get(podName, metav1.GetOptions{})
```

Comment:
Sleep?
Also, does it scale to run this once per instance?

Reply:
It def scales to run it once per instance - not sure if we need a pool of clients though (as we have in `testground` which makes thousands such calls), but since we have no more than 100 pods per sidecar, I think we are fine, unless we notice this to get slower. I tested with 30 pods on an instance, and the call was instant.

---

I will add a time.Sleep!

Reply:
Ah, so it doesn't call a _central_ server?

Reply:
Not taking a context here is a bit annoying but not the end of the world.

Reply:
It looks like the next, unreleased version of this API supports contexts.

Reply:
It is calling the `kube-apiserver`, which we could scale horizontally if we need to at some point (and these calls would then be load balanced).